core-bundle/tests/Functional/MetaDataTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Functional;
+
+use Contao\ContentModel;
+use Contao\CoreBundle\File\MetaData;
+use Contao\FilesModel;
+use Contao\System;
+use ReflectionClass;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class MetaDataTest extends KernelTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+
+        System::setContainer(self::bootKernel()->getContainer());
+
+        $GLOBALS['TL_LANG']['MSC']['deleteConfirmFile'] = '';
+    }
+
+    public function testCreatesMetaDataContainerFromContentModel(): void
+    {
+        /** @var ContentModel $model */
+        $model = (new ReflectionClass(ContentModel::class))
```

Comment:
Models can be created from array since Contao 4.7 (da68f4f44f6c73764e7f1fee6cc2b6c075909664)

Reply:
But the constructor does way more stuff than setting the data (which would need more setup to get it working).